var/spack/repos/builtin/packages/slepc/package.py

```diff
@@ -96,7 +96,7 @@ def install(self, spec, prefix):
 
         if '+arpack' in spec:
             options.extend([
-                '--with-arpack-dir=%s' % spec['arpack-ng'].prefix.lib,
+                '--with-arpack-dir=%s' % spec['arpack-ng'].prefix,
```

Comment:
perhaps its better to use [untested]
```
'--with-arpack-include=%s' % spec['arpack-ng'].prefix.include,
'--with-arpack-lib=%s' % spec['arpack-ng'].prefix.lib,
```

Reply:
well the issue i was seeing is, on rhel, spec['arpack-ng'].prefix.lib came out to ...lib even though the actual directory is lib64 on rhel.

Reply:
the documentation i linked to in the original PR description said that --with-arpack-dir should point to the arpack prefix, not a lib directory.

Reply:
Ok. So the primary bug is in the code that provides `arpack.prefix.lib`? so in arpack?

generally package-include/lib is preferred over package-dir option - this avoids some of the guesswork configure does [which can sometimes be wrong]

Reply:
@balay The option `--with-arpack-lib=` is intended to provide a list of libraries, not a path. Isn't `arpack.prefix.lib` a path?

Reply:
Ah - sorry - should have used:
```
'--with-arpack-lib=%s' % spec['arpack-ng'].prefix.libs.joined()
```

Reply:
It looks like `--with-arpack-include` is not recognized. At least that's what I'm seeing when I try to use it:

```
$> spack install -f slepc
...
==> Installing slepc-3.14.1-oa2mlhnzrwmvtkr45ii7qmffkxp4t73t
==> Using cached archive: /opt/spack/var/spack/cache/_source-cache/archive/cc/cc78a15e34d26b3e6dde003d4a30064e595225f6185c1975bbd460cb5edd99c7.tar.gz
==> slepc: Executing phase: 'install'
==> Error: ProcessError: Command exited with status 1:
    '/opt/spack/opt/spack/linux-rhel7-x86_64/gcc-8.1.0/python-3.8.6-e7dxrymceexhq6cg6f3jtp6sevyozzko/bin/python3.8' 'configure' '--prefix=/opt/spack/opt/spack/linux-rhel7-x86_64/gcc-8.1.0/slepc-3.14.1-oa2mlhnzrwmvtkr45ii7qmffkxp4t73t' '--with-arpack-include=/opt/spack/opt/spack/linux-rhel7-x86_64/gcc-8.1.0/arpack-ng-3.8.0-hu32xolzvdqbjdtqozml4e4bgmjkwhbz/include' '--with-arpack-lib=-lparpack,-larpack'

1 error found in build log:
     1    ==> slepc: Executing phase: 'install'
     2    ==> [2020-12-18-18:23:29.750450] '/opt/spack/opt/spack/linux-rhel7-x86_64/gcc-8.1.0/python-3.8.6-e7dxrymceexhq6cg6f3j
          tp6sevyozzko/bin/python3.8' 'configure' '--prefix=/opt/spack/opt/spack/linux-rhel7-x86_64/gcc-8.1.0/slepc-3.14.1-oa2m
          lhnzrwmvtkr45ii7qmffkxp4t73t' '--with-arpack-include=/opt/spack/opt/spack/linux-rhel7-x86_64/gcc-8.1.0/arpack-ng-3.8.
          0-hu32xolzvdqbjdtqozml4e4bgmjkwhbz/include' '--with-arpack-lib=-lparpack,-larpack'
  >> 3    Checking environment... ERROR: Invalid arguments --with-arpack-include=/opt/spack/opt/spack/linux-rhel7-x86_64/gcc-8.
          1.0/arpack-ng-3.8.0-hu32xolzvdqbjdtqozml4e4bgmjkwhbz/include
     4    Use -h for help
```

Reply:
Yes, this option is not available because in SLEPc we do not use ARPACK's header files. Probably we should ignore it instead of erroring out. Try with `--with-arpack-lib=` only. Thanks.

Reply:
In that case, I will merge this PR then. Everything about slepc stays the same except `--with-arpack-dir` is *correctly* set to the arpack install prefix. The current package sets `--with-arpack-dir` to the lib dir of arpack, which is not right. 

@balay do you object to this? We can always change to `--with-arpack-include` if, in the future, that option is supported.

Reply:
For reference, I have made changes in SLEPc that will be included in version 3.14.2, cf. [MR](https://gitlab.com/slepc/slepc/-/merge_requests/108).

Reply:
Looks like --with-arpack-include option does not exist - so only change would be to use --with-arpack-lib. However I see some extra code in slepc/package.py - that lists the arpack libraries separately - that can likely be removed.

I'm fine with this change for now.